controllers/metal3.io/host_state_machine.go

```diff
@@ -301,7 +301,8 @@ func (hsm *hostStateMachine) handleRegistering(info *reconcileInfo) actionResult
 }
 
 func (hsm *hostStateMachine) handleInspecting(info *reconcileInfo) actionResult {
-	actResult := hsm.Reconciler.actionInspecting(hsm.Provisioner, info)
+	force := hasInspectAnnotation(hsm.Host)
```

Comment:
* Isn't missing a check on [externally provisioned hosts](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/re-introspection-interface.md#proposal)?
* Evaluation of the flag couldn't be moved directly within the `actionInspecting`?

Reply:
By definition here we're in the Inspecting state, not the ExternallyProvisioned state.
I tend to agree with you that it would make sense to do this inside actionInspecting though.

Reply:
Right I was wrong, I misread the ExternallyProvisioned -> Inspecting path in the diagram not considering that couldn't happen when `spec.ExternallyProvisioned` is set.

Reply:
@zaneb @andfasano thanks a lot for your reviews. I have updated the patch now, and added refresh flag. Please take look.